bindings/wasm/output.go

```diff
@@ -101,11 +90,16 @@ func (out *outputBinding) Init(ctx context.Context, metadata bindings.Metadata)
 }
 
 func (out *outputBinding) Invoke(ctx context.Context, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	guestName := out.meta.GuestName
+	if guestName == "" {
+		guestName = out.module.Name()
+	}
+
 	// Currently, concurrent modules can conflict on name. Make sure we have
 	// a unique one.
 	instanceNum := out.instanceCounter.Add(1)
-	instanceName := out.guestName + "-" + strconv.FormatUint(instanceNum, 10)
-	moduleConfig := out.moduleConfig.WithName(instanceName)
+	instanceName := guestName + "-" + strconv.FormatUint(instanceNum, 10)
+	moduleConfig := wasm.NewModuleConfig(out.meta).WithName(instanceName)
```

Comment:
I think the config should be per-component and not per-invocation IMHO

Reply:
you mean to cache the `wasm.NewModuleConfig(out.meta)` thing as a field? can to. the other parts are per-invocation.

Reply:
It is configured on the component level(current by metadata and this may need to change depending on how we decide to pass this configuration). Because wasm binding creates a new wasm instance for every request, so the config needs to be used in Invoke method.

Reply:
yeah what @Taction said is the better answer. the base config from meta can be moved to a field, but it would really likely be moved back later